src/components/badge/IOBadge.tsx

```diff
@@ -0,0 +1,154 @@
+import React from "react";
+import { PixelRatio, Platform, StyleSheet, Text, View } from "react-native";
+import { IOColors } from "../../core/IOColors";
+import { makeFontStyleObject } from "../../utils/fonts";
+
+export type IOBadgeOutlineColors = Extract<
+  IOColors,
+  "blue" | "white" | "red" | "orange"
+>;
+export type IOBadgeSolidColors = Extract<
+  IOColors,
+  "blue" | "white" | "grey" | "aqua" | "red"
+>;
+
+type IOBadgeCommonProps = {
+  text: string;
+  small?: boolean;
+  labelColor?: Extract<IOColors, "bluegreyDark" | "blue" | "white" | "red">;
+  testID?: string;
+  labelTestID?: string;
+};
+
+type IOBadgeConditionalProps =
+  | {
+    variant: "solid";
+    color: IOBadgeSolidColors;
+  }
+  | {
+    variant: "outline";
+    color: IOBadgeOutlineColors;
+  };
+
+export type IOBadge = IOBadgeCommonProps & IOBadgeConditionalProps;
+
+type SolidVariantProps = {
+  background: IOColors;
+  text: IOColors;
+};
+
+const mapOutlineColor: Record<NonNullable<IOBadgeOutlineColors>, IOColors> = {
+  blue: "blue",
+  white: "white",
+  red: "red",
+  orange: "orange"
+};
+
+const mapSolidColor: Record<
+  NonNullable<IOBadgeSolidColors>,
+  SolidVariantProps
+> = {
+  blue: {
+    background: "blue",
+    text: "white"
+  },
+  white: {
+    background: "white",
+    text: "bluegrey"
+  },
+  aqua: {
+    background: "aqua",
+    text: "bluegreyDark"
+  },
+  grey: {
+    background: "grey",
+    text: "white"
+  },
+  red: {
+    background: "red",
+    text: "white"
+  }
+};
+
+const defaultVariant: IOBadge["variant"] = "solid";
+const defaultColor: IOBadge["color"] = "blue";
+
+const styles = StyleSheet.create({
+  badge: {
+    alignItems: "center",
+    justifyContent: "center",
+    ...Platform.select({
+      android: {
+        textAlignVertical: "center"
+      }
+    }),
+    // Visual parameters based on the FontScale
+    // ~20 = Small size height
+    // ~24 = Default size height
+    paddingVertical: PixelRatio.getFontScale() * 1.25,
+    paddingHorizontal: PixelRatio.getFontScale() * 10,
+    borderRadius: PixelRatio.getFontScale() * 25
+  },
+  label: {
+    alignSelf: "center",
+    ...makeFontStyleObject("SemiBold")
+  },
+  labelSizeDefault: {
+    fontSize: 14,
+    lineHeight: 20
+  },
+  labelSizeSmall: {
+    fontSize: 11,
+    lineHeight: 16
+  }
+});
+
+/**
+ * Official badge component
+@deprecated("Use the new Badge component instead")
```

Comment:
This `Badge` component was developed as an official replacement for the legacy NativeBase Badge component. But visually it's the same as the legacy one. Are we sure to want to keep it in the new design system?

Reply:
Yes, you're right. 
I've removed the component and the Design System section.
Thanks for the hint.